lib/viewmodels/home_view_model.dart

```diff
@@ -6,60 +8,73 @@ import "package:proxima/services/geolocation_service.dart";
     final postRepository = ref.watch(postRepositoryProvider);
     final userRepository = ref.watch(userRepositoryProvider);
 
     final position = await geoLocationService.getCurrentPosition();
 
-    final postsFirestore =
-        await postRepository.getNearPosts(position, kmPostRadius);
-
-    final postOwnersId =
-        postsFirestore.map((post) => post.data.ownerId).toSet();
+    final postsFirestoreStream =
+        postRepository.getNearPosts(position, kmPostRadius);
 
-    final postOwners = await Future.wait(
-      postOwnersId.map((userId) => userRepository.getUser(userId)),
-    );
+    final futurePostOverviewsStream =
+        postsFirestoreStream.map((postList) async {
+      final postOwnersId = postList.map((post) => post.data.ownerId).toSet();
 
-    final posts = postsFirestore.map((post) {
-      final owner = postOwners.firstWhere(
-        (user) => user.uid == post.data.ownerId,
-        // This should never be executed in practice as if the owner is not found,
-        // the user repository would have already thrown an exception.
-        orElse: () => throw Exception("Owner not found"),
+      final postOwners = await Future.wait(
+        postOwnersId.map((userId) => userRepository.getUser(userId)),
       );
 
-      return PostOverview(
-        title: post.data.title,
-        description: post.data.description,
-        voteScore: post.data.voteScore,
-        ownerDisplayName: owner.data.displayName,
-        commentNumber:
-            0, // TODO: Update appropriately when comments are implemented
-      );
-    }).toList();
+      final posts = postList.map((post) {
+        final owner = postOwners.firstWhere(
+          (user) => user.uid == post.data.ownerId,
+          // This should never be executed in practice as if the owner is not found,
+          // the user repository would have already thrown an exception.
+          orElse: () => throw Exception("Owner not found"),
+        );
+
+        return PostOverview(
+          title: post.data.title,
+          description: post.data.description,
+          voteScore: post.data.voteScore,
+          ownerDisplayName: owner.data.displayName,
+          commentNumber:
+              0, // TODO: Update appropriately when comments are implemented
+        );
+      }).toList();
+
+      return posts;
+    });
 
-    return posts;
+    await for (final futurePostOverviews in futurePostOverviewsStream) {
+      yield await futurePostOverviews;
+    }
   }
 
-  /// Refresh the list of posts
-  /// This will put the state of the viewmodel to loading, fetch the posts
-  /// and update the state accordingly
-  Future<void> refresh() async {
-    state = const AsyncValue.loading();
-    state = await AsyncValue.guard(() => build());
+  /// Refreshes the list of posts to display the posts that are near the new location of the user
+  /// This method should be called whenever the location of the user changes.
+  void refresh() {
```

Comment:
We might want to do something closer to the documentation here, see https://riverpod.dev/docs/essentials/side_effects#using-refinvalidateself-to-refresh-the-provider.

Maybe just calling `ref.invalidateSelf(); await future;` could work ?